string input_str
---
string segment_status